app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -216,6 +216,9 @@ def track_events
         sign_in_flow: session[:sign_in_flow],
         vtr: sp_session[:vtr],
         acr_values: sp_session[:acr_values],
+        sign_in_duration_seconds: (Time.zone.now - Time.zone.parse(
+          session[:sign_in_page_visited_at] || Time.zone.now.to_s,
+        )).seconds.to_i.round(2),
```

Comment:
1. if the `session[:sign_in_page_visited_at]` gets cleared or wiped, this will be zero because it's `.now`. What if we left it `nil` when is no `session[:sign_in_page_visited_at]` so that we can tell the difference between missing data and zero?

2. `.to_i.round(2)` is redundant, because integers can't be rounded. Is the goal to have 2 decimals of floating point precision?

Reply:
1. That's a good point. It might also be helpful to get an idea of how often someone is bouncing to a different browser so instead of the data showing a huge spike it will have some nil values to query against. 🤔 

2. I didn't realize converting to integer already truncates https://dev.to/dawncronin/number-conversions-in-ruby-1ned#:~:text=Ruby%20has%20a%20built%20in,will%20round%20to%20two%20decimals. "_the to_i conversion truncates the number,_"